cloud/blockstore/tests/csi_driver/test.py

```diff
@@ -346,20 +378,26 @@ def test_nbs_csi_driver_volume_stat():
         cleanup_after_test(env)
 
 
-@pytest.mark.parametrize('mount_path,volume_access_type',
-                         [("/var/lib/kubelet/pods/123/volumes/kubernetes.io~csi/456/mount", "mount"),
-                          ("/var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish/123/456", "block")])
-def test_csi_sanity_nbs_backend(mount_path, volume_access_type):
-    env, run = init()
+@pytest.mark.parametrize('mount_path,volume_access_type,vm_mode',
+                         [("/var/lib/kubelet/pods/123/volumes/kubernetes.io~csi/456/mount", "mount", False),
+                          ("/var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish/123/456", "block", False),
+                          ("/var/lib/kubelet/pods/123/volumes/kubernetes.io~csi/456/mount", "mount", True)])
+def test_csi_sanity_nbs_backend(mount_path, volume_access_type, vm_mode):
+    env, run = init(vm_mode)
     backend = "nbs"
 
     try:
         CSI_SANITY_BINARY_PATH = common.binary_path("cloud/blockstore/tools/testing/csi-sanity/bin/csi-sanity")
         mount_dir = Path(mount_path)
         mount_dir.parent.mkdir(parents=True, exist_ok=True)
 
+        volume_context = {
+            "backend": backend,
+            "instanceId": "test-instance-id",
+        }
         params_file = Path(os.getcwd()) / "params.yaml"
-        params_file.write_text(f"backend: {backend}")
+        with open(params_file, 'w') as f:
```

Comment:
with params_file.open('w') or even better params_file.write_text(yaml.safe_dump(volume_context)). And you can inline volume_context then. All that can be made into a one logical line. Also, Path("params.yaml") would achieve the same result as os.getcwd() / "params.yaml", so the code can be simplified even more.